File: QE-Selenium-Java/Activity11.java
package demos;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activiy11{
    public static void main(String[] args) {

        FirefoxDriver driver = new FirefoxDriver();
        Actions actions = new Actions(driver);

        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.get("https://training-support.net/webelements/dynamic-controls");

        System.out.println(driver.getTitle());

        driver.findElement(By.id("checkbox"));

        WebElement toggleButton = driver.findElement(By.xpath("//button[contains(text(), 'Toggle Checkbox')]"));

        actions.click(toggleButton).build().perform();

        wait.until(ExpectedConditions.invisibilityOfElementLocated(By.id("checkbox")));

        System.out.println("waited until checkbox was disappeared");

        actions.click(toggleButton).build().perform();

        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("checkbox")));

        System.out.println("waited until checkbox was visible");
        driver.quit();
    }
}

File: QE-Selenium-Python/Activity11.py
from selenium import webdriver

from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.common.action_chains import ActionChains

driver = webdriver.Firefox()
actions = webdriver.ActionChains(driver)

wait = WebDriverWait(driver, 10)

driver.get("https://training-support.net/webelements/dynamic-controls")

print(driver.title)

driver.find_element(By.ID, "checkbox")

toggleButton = driver.find_element(By.XPATH,"//button [contains(text(), 'Toggle Checkbox')]")

actions.click(toggleButton).perform()

wait.until(EC.invisibility_of_element_located ((By.ID, "checkbox")))

print("waited until checkbox was disappeared")

actions.click(toggleButton).perform()

wait.until(EC.visibility_of_element_located((By.ID, "checkbox")))

print("waited until checkbox was visible")
driver.quit()